### PR TITLE
improved code quality

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: interfaces/bots/interface_bot.py [hunk 1]
```diff
@@ -196,32 +196,32 @@ def get_command_portfolio():
...
                 .format(
-                REAL_TRADER_STR,
-                PrettyPrinter.portfolio_profitability_pretty_print(real_global_profitability,
-                                                                   None,
-                                                                   get_reference_market()),
-                PrettyPrinter.get_min_string_from_number(real_percent_profitability, 2),
-                PrettyPrinter.get_min_string_from_number(market_average_profitability, 2),
-                PrettyPrinter.get_min_string_from_number(real_no_trade_profitability, 2),
-                EOL)
+                    REAL_TRADER_STR,
```
Comment (Herklos) [inline interfaces/bots/interface_bot.py]: Why keeping `format` here ? and not using f-string

Comment (GuillaumeDSM) [reply interfaces/bots/interface_bot.py]: right

File: config/config.py [hunk 2]
```diff
@@ -26,25 +26,25 @@
 
 def load_config(config_file=CONFIG_FILE, error=True):
     logger = logging.getLogger("CONFIG LOADER")
-    basic_error = "Error when load config file {0}".format(config_file)
+    basic_error = f"Error when load config file {config_file}"
```
Comment (Herklos) [inline config/config.py]: I don't think f-string migration on this file is a good idea. It's the first module been imported by start.py at startup, before `tools/__init__.py` that check python compatibility.
So i suggest to revert this modifications **OR** import `tools/__init__.py` before.

File: tools/tentacle_manager/tentacle_package_manager.py [hunk 3]
```diff
@@ -138,11 +136,12 @@ def process_module(self, action, package, module_name, package_localisation, is_
 
                 if module_test_files:
                     for test in module_tests:
-                        test_loc = "{0}.py".format(TentacleUtil.create_localization_from_type(package_localisation,
-                                                                                              module_type,
-                                                                                              module_subtype,
-                                                                                              test,
-                                                                                              True))
+                        location_str = TentacleUtil.create_localization_from_type(package_localisation,
```
Comment (Herklos) [inline tools/tentacle_manager/tentacle_package_manager.py]: Why did you need to split this line up ?

File: tools/tentacle_manager/tentacle_util.py [hunk 4]
```diff
@@ -207,40 +207,27 @@ def create_localization_from_type(localization, module_type, module_subtype, fil
...
     if tests:
-        test_folder_if_required = "/{0}".format(TENTACLES_TEST_PATH)
+        test_folder_if_required = f"/{TENTACLES_TEST_PATH}"
     if module_subtype:
-        return "{0}{1}/{2}/{3}/{4}".format(TENTACLES_PATH,
-                                           test_folder_if_required,
-                                           TENTACLE_TYPES[module_type],
-                                           TENTACLE_TYPES[module_subtype],
-                                           target_folder)
+        return f"{TENTACLES_PATH}{test_folder_if_required}/{TENTACLE_TYPES[module_type]}/" \
```
Comment (Herklos) [inline tools/tentacle_manager/tentacle_util.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here.

File: tools/tentacle_manager/tentacle_package_manager.py [hunk 5]
```diff
@@ -69,19 +69,16 @@ def _process_action_on_module(self, action, module_type, module_subtype,
...
 
             # Update local __init__
-            line_in_init = "from .{0} import *\n".format(module_name)
-            init_file = "{0}/{1}/{2}/{3}/{4}".format(TENTACLES_PATH,
-                                                     TENTACLE_TYPES[module_type],
-                                                     TENTACLE_TYPES[module_subtype],
-                                                     target_folder,
-                                                     PYTHON_INIT_FILE)
+            line_in_init = f"from .{module_name} import *\n"
+            init_file = f"{TENTACLES_PATH}/{TENTACLE_TYPES[module_type]}/{TENTACLE_TYPES[module_subtype]}/" \
```
Comment (Herklos) [inline tools/tentacle_manager/tentacle_package_manager.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: tools/tentacle_creator/tentacle_creator.py [hunk 6]
```diff
@@ -36,21 +36,13 @@ def __init__(self, config):
...
 
     @staticmethod
     def get_config_template_path(name):
-        return "{0}/{1}/{2}/{3}{4}{5}".format(TOOLS_PATH,
-                                              TENTACLE_CREATOR_PATH,
-                                              TENTACLE_TEMPLATE_PATH,
-                                              name,
-                                              TENTACLE_CONFIG_TEMPLATE_PRE_EXT,
-                                              TENTACLE_TEMPLATE_EXT)
+        return f"{TOOLS_PATH}/{TENTACLE_CREATOR_PATH}/{TENTACLE_TEMPLATE_PATH}/{name}" \
```
Comment (Herklos) [inline tools/tentacle_creator/tentacle_creator.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: tools/tentacle_manager/tentacle_package_manager.py (hunk 3 above)
Comment (GuillaumeDSM) [reply tools/tentacle_manager/tentacle_package_manager.py]: readability

File: tools/tentacle_creator/tentacle_creator.py [hunk 7]
```diff
@@ -36,21 +36,13 @@ def __init__(self, config):
...
     @staticmethod
     def get_template_path(name):
-        return "{0}/{1}/{2}/{3}{4}{5}".format(TOOLS_PATH,
-                                              TENTACLE_CREATOR_PATH,
-                                              TENTACLE_TEMPLATE_PATH,
-                                              name,
-                                              TENTACLE_TEMPLATE_PRE_EXT,
-                                              TENTACLE_TEMPLATE_EXT)
+        return f"{TOOLS_PATH}/{TENTACLE_CREATOR_PATH}/{TENTACLE_TEMPLATE_PATH}/{name,}{TENTACLE_TEMPLATE_PRE_EXT}" \
+            f"{TENTACLE_TEMPLATE_EXT}"
```
Comment (Herklos) [inline tools/tentacle_creator/tentacle_creator.py]: Same here

File: tools/pretty_printer.py [hunk 8]
```diff
@@ -55,39 +50,28 @@ def trade_pretty_printer(trade):
...
+            f"Result : {str(result).split('.')[1]}\n Evaluation : {final_eval}"
 
     @staticmethod
     def global_portfolio_pretty_print(global_portfolio, separator="\n"):
-        result = ["{0} ({1}) {2}".format(
-            PrettyPrinter.get_min_string_from_number(amounts[Portfolio.TOTAL]),
-            PrettyPrinter.get_min_string_from_number(amounts[Portfolio.AVAILABLE]),
-            currency)
-            for currency, amounts in global_portfolio.items() if amounts[Portfolio.TOTAL] > 0]
+        result = [f"{PrettyPrinter.get_min_string_from_number(amounts[Portfolio.TOTAL])} " \
```
Comment (Herklos) [inline tools/pretty_printer.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: tools/pretty_printer.py [hunk 9]
```diff
@@ -55,39 +50,28 @@ def trade_pretty_printer(trade):
...
+            f"{convert_timestamp_to_datetime(trade.filled_time)} "
 
     @staticmethod
     def cryptocurrency_alert(crypto_currency, symbol, result, final_eval):
-        return "OctoBot ALERT : #{0}\n Symbol : #{1}\n Result : {2}\n Evaluation : {3}".format(
-            crypto_currency,
-            symbol.replace("/", ""),
-            str(result).split(".")[1],
-            final_eval)
+        return f"OctoBot ALERT : #{crypto_currency}\n Symbol : #{symbol.replace('/', '')}\n " \
```
Comment (Herklos) [inline tools/pretty_printer.py]: Same here

File: tools/pretty_printer.py [hunk 10]
```diff
@@ -55,39 +50,28 @@ def trade_pretty_printer(trade):
...
-        return "{0}: {1} {2} at {3} {4} on {5}: {6} ".format(
-            order_type_name,
-            PrettyPrinter.get_min_string_from_number(trade.quantity),
-            currency,
-            PrettyPrinter.get_min_string_from_number(trade.price),
-            market,
-            trade.exchange.get_name(),
-            convert_timestamp_to_datetime(trade.filled_time))
+        return f"{order_type_name}: {PrettyPrinter.get_min_string_from_number(trade.quantity)} {currency} " \
+            f"at {PrettyPrinter.get_min_string_from_number(trade.price)} {market} on {trade.exchange.get_name()}: " \
```
Comment (Herklos) [inline tools/pretty_printer.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: tools/pretty_printer.py [hunk 11]
```diff
@@ -33,14 +33,9 @@ def open_order_pretty_printer(order):
...
-        return "{0}: {1} {2} at {3} {4} on {5}: {6} ".format(
-            order_type_name,
-            PrettyPrinter.get_min_string_from_number(order.get_origin_quantity()),
-            currency,
-            PrettyPrinter.get_min_string_from_number(order.get_origin_price()),
-            market,
-            order.get_exchange().get_name(),
-            convert_timestamp_to_datetime(order.get_creation_time()))
+        return f"{order_type_name}: {PrettyPrinter.get_min_string_from_number(order.get_origin_quantity())} " \
+            f"{currency} at {PrettyPrinter.get_min_string_from_number(order.get_origin_price())} {market} on " \
```
Comment (Herklos) [inline tools/pretty_printer.py]: Same here

File: interfaces/bots/interface_bot.py [hunk 12]
```diff
@@ -167,26 +164,16 @@ def get_command_portfolio():
...
-            portfolios_string += "{0}Portfolio value : {1} {2}{3}".format(
-                REAL_TRADER_STR,
-                PrettyPrinter.get_min_string_from_number(portfolio_real_current_value),
-                reference_market,
-                EOL)
-            portfolios_string += "{0}Portfolio : {2}{1}{2}{2}".format(
-                REAL_TRADER_STR,
-                PrettyPrinter.global_portfolio_pretty_print(real_global_portfolio),
-                EOL)
+            portfolios_string += f"{REAL_TRADER_STR}Portfolio value : " \
```
Comment (Herklos) [inline interfaces/bots/interface_bot.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: evaluator/abstract_evaluator.py [hunk 13]
```diff
@@ -62,8 +62,8 @@ def get_all_subclasses(cls):
 
     @classmethod
     def get_config_file_name(cls, config_evaluator_type=None):
-        return "{0}/{1}/{2}/{3}/{4}".format(TENTACLES_PATH, TENTACLES_EVALUATOR_PATH, config_evaluator_type
-                                            , EVALUATOR_CONFIG_FOLDER, cls.get_name() + CONFIG_FILE_EXT)
+        return f"{TENTACLES_PATH}/{TENTACLES_EVALUATOR_PATH}/{config_evaluator_type}/{EVALUATOR_CONFIG_FOLDER}" \
```
Comment (Herklos) [inline evaluator/abstract_evaluator.py]: I think that a [_Multi-line string syntax_](https://www.smallsurething.com/multi-line-strings-in-python/) will be better here. (with `"""` )

File: tools/tentacle_manager/tentacle_package_manager.py (hunk 3 above)
Comment (Herklos) [reply tools/tentacle_manager/tentacle_package_manager.py]: Ok why not

File: tools/__init__.py [hunk 14]
```diff
@@ -20,7 +20,7 @@
 
 from config import PlatformsName, get_os
 
-MIN_PYTHON_VERSION = (3, 6)
+MIN_PYTHON_VERSION = (3, 7)
```
Comment (Herklos) [inline tools/__init__.py]: Yes !
